packages/@ember/-internals/runtime/lib/mixins/array.ts

```diff
@@ -1775,41 +1929,45 @@ let NativeArray = Mixin.create(MutableArray, Observable, {
 // Remove any methods implemented natively so we don't override them
 const ignore = ['length'];
 NativeArray.keys().forEach((methodName) => {
-  if (Array.prototype[methodName]) {
+  // SAFETY: It's safe to read unknown properties from an object
+  if ((Array.prototype as any)[methodName]) {
     ignore.push(methodName);
   }
 });
 
 NativeArray = NativeArray.without(...ignore);
 
-let A;
+let A: <T>(arr?: Array<T>) => NativeArray<T>;
```

Comment:
Previous types were too permissive here for the value of `arr`.

Reply:
It *is* generic! At the other site, can you just `= A<T>()`? (What a mess of characters that is, though 😂)

Reply:
Yeah, I fixed the other spot.